Backend/Controllers/Members.cs

```diff
@@ -0,0 +1,513 @@
+        /// </summary>
+        /// <param name="id">The id of the member to update.</param>
+        /// <param name="newBegunstiger">The new begunstiger status for the member.</param>
+        /// <returns>No content.</returns>
+        [HttpPatch("{id}/begunstiger")]
+        public async Task<IActionResult> PatchMemberBegunstiger(uint id, bool newBegunstiger, CancellationToken cancellationToken)
+        {
+            Member? member = await db.Members.FindAsync(id, cancellationToken);
+            if (member == null) return NotFound();
+
+            member.Begunstiger = newBegunstiger;
+
+            await db.SaveChangesAsync(cancellationToken);
+
+            return NoContent();
+        }
+
+        // PATCH: api/members/5/suspended
+        /// <summary>
+        /// Updates a member's suspended status.
+        /// </summary>
+        /// <param name="id">The id of the member to update.</param>
+        /// <param name="newSuspended">The new suspended status for the member.</param>
+        /// <returns>No content.</returns>
+        [HttpPatch("{id}/suspended")]
+        public async Task<IActionResult> PatchMemberSuspended(uint id, bool newSuspended, CancellationToken cancellationToken)
+        {
+            Member? member = await db.Members.FindAsync(id, cancellationToken);
+            if (member == null) return NotFound();
+
+            member.Suspended = newSuspended;
+
+            await db.SaveChangesAsync(cancellationToken);
+
+            return NoContent();
+        }
+
+        // PUT: api/members/5
+        /// <summary>
+        /// Updates a member's details.
+        /// </summary>
+        /// <param name="id">The id of the member to update.</param>
+        /// <param name="memberDto">The new details of the member.</param>
+        /// <returns>The updated member.</returns>
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutMember(uint id, MemberUpdateDTO memberDto, CancellationToken cancellationToken)
+        {
+            Member? member = await db.Members.FindAsync(id, cancellationToken);
+            if (member == null) return NotFound();
+
+            member.StudentNumber = memberDto.StudentNumber;
+            member.FirstName = memberDto.FirstName;
+            member.LastName = memberDto.LastName;
+            member.Email = memberDto.Email;
+            member.PhoneNumber = memberDto.PhoneNumber;
+            member.Address = memberDto.Address;
+            member.DateOfBirth = memberDto.DateOfBirth;
+            member.PreferredLanguage = memberDto.PreferredLanguage;
+            member.Notes = memberDto.Notes;
+            member.Gratie = memberDto.Gratie;
+            member.LidVanVerdienste = memberDto.LidVanVerdienste;
+            member.EreLid = memberDto.EreLid;
+            member.Begunstiger = memberDto.Begunstiger;
+            member.Suspended = memberDto.Suspended;
+
+            await db.SaveChangesAsync(cancellationToken);
+
+            return NoContent();
```

Comment:
This is bascially a large PATCH function. PUT implied a full replacement. This raises the issue here: what if in the future we add an extra field? This code will still compile, thus 'forgetting' to allow to replace that field as well. If instead of a DTO you accept a full type here, _then_ you are actually serving as a PUT: whatever happens, you request a full model and replace a full model (just check if the primary key matches). Now this is just a large PATCH.